subscription.go

```diff
@@ -166,7 +166,15 @@ func (r *marathonClient) registerSSESubscription() error {
 	url := fmt.Sprintf("%s/%s", marathon, marathonAPIEventStream)
 
 	// Try to connect to stream
-	stream, err := eventsource.Subscribe(url, "")
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return err
+	}
+	if r.config.HTTPBasicAuthUser != "" {
```

Comment:
I'm not completely sure if HTTP Basic Auth supports empty usernames with non-empty passwords. When in doubt, we should probably extend the if condition by `|| r.config.HTTPBasicPassword != ""`.